modules/library/main/src/main/java/org/geotools/feature/collection/DecoratingClippingFeatureCollection.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.feature.collection;
+
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureIterator;
+
+/**
+ * The class decorates a SimpleFeatureCollection with one that returns features that intersect the
+ * geometry passed with a Hints.GEOMETRY_CLIP
+ */
+public class DecoratingClippingFeatureCollection extends DecoratingSimpleFeatureCollection {
```

Comment:
Thanks for making this a general functionality.

```suggestion
public class ClippingFeatureCollection extends DecoratingSimpleFeatureCollection {
```

None of the existing decorators reminds you that it's a decorator in the name:

![image](https://user-images.githubusercontent.com/325433/81175529-fc169100-8fa3-11ea-932e-8e7703303794.png)